daemon/src/taker.rs

```diff
@@ -292,3 +242,93 @@ async fn main() -> Result<()> {
 
     Ok(())
 }
+
+pub struct ActorSystem<O, M> {
+    cfd_actor_addr: Address<taker_cfd::Actor<O, M>>,
+    cfd_feed_receiver: watch::Receiver<Vec<Cfd>>,
+    order_feed_receiver: watch::Receiver<Option<Order>>,
+    update_cfd_feed_receiver: watch::Receiver<UpdateCfdProposals>,
+}
+
+impl<O, M> ActorSystem<O, M>
+where
+    O: xtra::Handler<oracle::MonitorAttestation>
+        + xtra::Handler<oracle::GetAnnouncement>
+        + xtra::Handler<oracle::FetchAnnouncement>
+        + xtra::Handler<oracle::Sync>,
+    M: xtra::Handler<monitor::StartMonitoring>
+        + xtra::Handler<monitor::Sync>
+        + xtra::Handler<monitor::CollaborativeSettlement>,
+{
+    pub async fn new<F>(
+        db: SqlitePool,
+        wallet: Wallet,
+        oracle_pk: schnorrsig::PublicKey,
+        send_to_maker: Box<dyn MessageChannel<wire::TakerToMaker>>,
+        read_from_maker: Box<dyn Stream<Item = taker_cfd::MakerStreamMessage> + Unpin + Send>,
+        oracle_constructor: impl Fn(Vec<Cfd>, Box<dyn StrongMessageChannel<Attestation>>) -> O,
+        monitor_constructor: impl Fn(Box<dyn StrongMessageChannel<monitor::Event>>, Vec<Cfd>) -> F,
+    ) -> Self
+    where
+        F: Future<Output = Result<M>>,
+    {
+        let mut conn = db.acquire().await.unwrap();
+
+        let cfds = load_all_cfds(&mut conn).await.unwrap();
+
+        let (cfd_feed_sender, cfd_feed_receiver) = watch::channel(cfds.clone());
+        let (order_feed_sender, order_feed_receiver) = watch::channel::<Option<Order>>(None);
+        let (update_cfd_feed_sender, update_cfd_feed_receiver) =
+            watch::channel::<UpdateCfdProposals>(HashMap::new());
+
+        let (monitor_addr, mut monitor_ctx) = xtra::Context::new(None);
+        let (oracle_addr, mut oracle_ctx) = xtra::Context::new(None);
+
+        let cfd_actor_addr = taker_cfd::Actor::new(
+            db,
+            wallet,
+            oracle_pk,
+            cfd_feed_sender,
+            order_feed_sender,
+            update_cfd_feed_sender,
+            send_to_maker,
+            monitor_addr,
+            oracle_addr,
+        )
+        .create(None)
+        .spawn_global();
+
+        tokio::spawn(cfd_actor_addr.clone().attach_stream(read_from_maker));
+
+        tokio::spawn(
+            monitor_ctx
+                .notify_interval(Duration::from_secs(20), || monitor::Sync)
+                .unwrap(),
+        );
+        tokio::spawn(
+            monitor_ctx.run(
+                monitor_constructor(Box::new(cfd_actor_addr.clone()), cfds.clone())
+                    .await
+                    .unwrap(),
+            ),
+        );
+
+        tokio::spawn(
+            oracle_ctx
+                .notify_interval(Duration::from_secs(5), || oracle::Sync)
+                .unwrap(),
+        );
+        let fan_out_actor = fan_out::Actor::new(&[&cfd_actor_addr, &cfd_actor_addr])
```

Comment:
:stop_sign: Why are we passing the same actor twice here? That can't be right!

Reply:
it was a bit weird for me too, but I copied it from already existing pattern:
https://github.com/comit-network/hermes/blob/taker-actor-system/daemon/src/maker.rs#L356

(shall I fix it both places? :) )

Reply:
done! I'll attach the maker commit to this PR as well.